src/plugins/intel_gna/src/transformations/pwl/segment_generator.cpp

```diff
@@ -0,0 +1,250 @@
+                                                 segments,
+                                                 lower_bound,
+                                                 upper_bound,
+                                                 subinterval.is_negative(),
+                                                 data_for_calculation->get_max_iterations());
+
+        } while (!validator.is_valid(error) && ++segments <= kMaxSegmentsNum);
+
+        IE_ASSERT(pwls.size() <= kMaxSegmentsNum)
+            << "Failed to converge part of activation_function(" << m_representation->get_name()
+            << ", lower_bound: " << lower_bound << ", upper_bound: " << upper_bound
+            << ", with allowed error percentage: " << data_for_calculation->get_allowed_error_percentage() << ")!\n";
+
+        if (subinterval.is_negative()) {
+            negative_pwls(pwls);
+        }
+
+        // Remove last segment from previous subinterval in case inserting subsequent subinterval
+        if (!total_pwls.empty()) {
+            total_pwls.pop_back();
+        }
+        total_pwls.insert(total_pwls.end(), pwls.begin(), pwls.end());
+    }
+
+    if (!total_pwls.empty()) {
+        auto inserter = data_for_calculation->get_surrounding_segments_inserter();
+        inserter->insert_surrounding_segments(total_pwls);
+    }
+
+    IE_ASSERT(total_pwls.size() <= kMaxSegmentsOverall)
+        << "Failed to converge whole activation_function(" << m_representation->get_name()
+        << ", lower_bound: " << overall_lower_bound << ", upper_bound: " << overall_upper_bound
+        << ", with allowed error percentage: " << data_for_calculation->get_allowed_error_percentage() << ")!\n";
+
+    return total_pwls;
+}
+
+std::shared_ptr<ov::Node> SegmentsGeneratorImpl::get_node() const {
+    return m_node;
+}
+
+std::pair<double, std::vector<Segment>> SegmentsGeneratorImpl::pivot_search(const Function& activation_function,
+                                                                            uint32_t segments_num,
+                                                                            double alpha_0,
+                                                                            double alpha_N,
+                                                                            bool negative,
+                                                                            size_t max_iterations) const {
+    std::vector<std::vector<double>> t(segments_num + 1);
+    std::vector<std::vector<double>> alpha(segments_num + 1);
+    std::vector<std::vector<double>> epsilon(segments_num + 1);
+    std::vector<std::vector<double>> d(segments_num + 1);
+    bool same_epsilon = false;
+    double Delta;
+    double epsilon_final = 0.0;
+    double max_epsilon = 0.0;
+    double max_epsilon_prev;
+    double min_epsilon;
+    double sgn = (negative) ? -1.0 : 1.0;
+    int j;
+
+    // Figure 4:  Box #1
+    j = 0;
+    Delta = 1.0;
+
+    for (int i = 0; i < segments_num; i++) {
```

Comment:
This will trigger a warning - comparison of signed and unsigned value.
Please also check other loops.